### DISCUSSION TITLE
Conditional values in ~/.config/mock.cfg

### DISCUSSION THREAD

**Discussion by brianjmurrell** [rpm-software-management/mock#1404] [Q&A]
Let's say I only want to add a value (a repo configuration for example) to `config_opts['dnf.conf']` if the target chroot is a particular value since that repo would only be valid for a given mock target configuration.

Is there any way to achieve this?

**Comment by xsuchy** [comment]
You have two options:

1) the config is evaluated using python function `eval()`. Therefore you can put there any python code.

2) The values are expanded using Jinja2 so you can use https://docs.rewst.help/documentation/jinja/common-jinja-examples/conditional-statements-and-logical-operators For caveats and examples see https://github.com/xsuchy/templated-dictionary (this is library that Mock uses for the evaluation).

**Comment by brianjmurrell** [comment]
So, I did see some Jinja2 templating in `/etc/mock/templates`, I was not aware that base `.cfg` files were templated also.

So I think the question becomes what values can be used in templates?  I.e. specifically for being able to apply settings conditionally on say the distro being built for in `mock`'s `-r` argument.
